### DISCUSSION TITLE
Not able to set up Authenticated Imgur Upload

### DISCUSSION THREAD

**Discussion by shiwer** [gavvvr/obsidian-imgur-plugin#36] [Getting help]
Hello,

I have been trying over and over again to setup authenticated imgur upload with no success.

I've been trying to set it up with multiple browsers (Brave, Chrome & Mozilla). 
Everytime it's the same, I go to the obsidian plugin. I click on the Authenticate button. It opens my browser. Then I write my creds and click allow. Then nothing seems to happen. Then I click again and I get a 403 from Imgur. 

When I go to my settings I see that I have allowed the Obsidian plugin but because it has not set up the token in my obs plugin, I'm not able to use it. So i disable the token and try again until I have a 429 that tells my I tried too many times and that I cannot set it up anymore.
![Screenshot from 2022-08-05 15-20-40](https://user-images.githubusercontent.com/22575583/183086993-24210f77-1ece-44dc-80d7-b6b93b1fb875.png)

**Comment by gavvvr** [comment]
Hi @shiwer 

> Then I write my creds and click allow. **Then nothing seems to happen.**

This is the problem. After you press "allow", your browser should redirect to Obsidian, but it does not. I can not say why, looks like none of your browsers perform redirect, or a handler for `obsidian://` is not registered in your system (browser probably tries to redirect, but nothing happens, because the system does not know how to handle it).

I've just tested the authentication on my machine and it works fine for me

**Comment by shiwer** [comment]
Thanks for closing but I still try to manage that issue. :s
Can you tell me how to set the token manually. I've tried on another computer it seems to be set. The  thing is I'm using Git Obsidian but I'm having issues making it work correctly  with Windows env. 
My Linux and Windows env are not sync.

I've been looking in the .obsidian\plugins\obsidian-imgur-plugin folder but I can't find an env file.

**Comment by gavvvr** [comment]
Hi @shiwer 
That's right, you won't find any tokens in `.obsidian\plugins\obsidian-imgur-plugin`. If it was stored in vault's `.obsidian`, the  Imgur session would not be shared between vaults and you would have to authenticate in each Obsidian vault.
I used [LocalStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) to persist OAuth token in a single place.

If you want to try some workaround, I can suggest you to open dev console and add the token manually...
<img width="232" alt="image" src="https://user-images.githubusercontent.com/1719646/184446150-fb9bde11-163d-4ba0-a3f1-c606d0c9afad.png">

Obtain the working token from your Linux machine and set it on the Windows machine where you have authentication problems.

For reference, see the following pieces of code:

https://github.com/gavvvr/obsidian-imgur-plugin/blob/e5260004a525b5249f77c6a39017c819ba65ce97/src/imgur/constants.ts#L3
https://github.com/gavvvr/obsidian-imgur-plugin/blob/e5260004a525b5249f77c6a39017c819ba65ce97/src/ui/ImgurPluginSettingsTab.ts#L38-L41
https://github.com/gavvvr/obsidian-imgur-plugin/blob/e5260004a525b5249f77c6a39017c819ba65ce97/src/uploader/imgUploaderFactory.ts#L20-L22

**Reply by shiwer** [reply]
Thanks for the workaround. I'll try it later that day !! ;)
